docs/testing.md

```diff
@@ -1,180 +1,67 @@
+1. Reopen the project in a dev container
+1. Open the terminal.
+1. Type "sudo su" to run commands as root
+1. Type ./build_and_run_unit_tests.sh to run the script
 
-1. Follow the [ODE installation instructions](https://github.com/usdot-jpo-ode/jpo-ode#documentation)
-1. Start a terminal for launching the ODE containers.
-1. Set the following environment variables:
+### Testing Using Docker
+1. Start by building the Docker image:
 
 ```bash
-$ export DOCKER_HOST_IP=<your.host.ip>
-$ export DOCKER_SHARED_VOLUME=<your.shared.directory>
+$ docker build -t ppm .
 ```
 
-1. Follow the Deploying ODE Application on a Docker Host directions
+2. Then run unit tests inside the container with the following command:
 
 ```bash
-$ docker-compose up --no-recreate -d
+$ docker run -it -e PPM_LOG_TO_CONSOLE=true --name ppm ppm /cvdi-stream-build/ppm_tests
 ```
 
-1. Start a terminal for launching the PPM.
+3. Remove the container:
 
 ```bash
-$ cd $BASE_PPM_DIR/jpo-cvdp/build
-$ ./ppm -c ../config/<testconfig>.properties
+$ docker rm ppm
 ```
-1. Open a web browser, and enter the url: `localhost:8080`
-1. Click on the **Connect** button.
-1. Click on the **Browse** button, find a JSON test file with BSMs (one per line).
-1. Click the **Upload** button.
-
-The BSMs from the file should be listed in the web browser: the BSMs section
-lists all the BSMs; the Filtered BSMs section contains the BSMs that were
-processed by the PPM and returned back to the ODE.
-
-### Testing without the ODE
 
-These instructions describe how to run a collection of BSM test JSON objects through the PPM and examine its operation.
-Using *GNU screen* for this work is really handy; you will need several shells.
-
-Startup `kafka-docker` in its own shell:
-
-```bash
-$ cd $BASE_PPM_DIR/kafka-docker
-$ docker-compose up --no-recreate -d                    // to startup kafka and zookeeper containers
-$ docker-compose ps                                     // to check that they are running.
+## Standalone Testing
+1. Spin up Kafka & the PPM
 ```
-
-In another shell, create the simulated ODE produced topic (`j2735BsmRawJson`) and PPM produced topic (`j2735BsmFilteredJson`)
-
-```bash
-$ cd $BASE_PPM_DIR/kafka
-$ bin/kafka-topic.sh --create --zookeeper <HOST IP>:2181 --replication-factor 1 --partitions 1 --topic j2735BsmRawJson
-$ bin/kafka-topic.sh --create --zookeeper <HOST IP>:2181 --replication-factor 1 --partitions 1 --topic j2735BsmFilteredJson
+$ docker compose up -d --build
 ```
 
-Startup the simulated ODE consumer in the same shell you used to create the topics.
-
-```bash
-$ cd $BASE_PPM_DIR/kafka                                
-$ bin/kafka-console-consumer.sh --bootstrap-server <HOST IP>:9092 --topic j2735BsmFilteredJson
+2. View logs of PPM
 ```
-
-- This process should just wait for input from the PPM module.
-
-In another shell, startup the PPM
-
-```bash
-$ cd $BASE_PPM_DIR/jpo-cvdp/build
-$ ./ppm -c ../config/<testconfig>.properties
+$ docker compose logs -f ppm
```

Comment:
It would be nice if we could provide detail instruction to rename sample.env and what shared volume folder to create and which relevant files to be moved to the shared volume folder.

Reply:
Detailed instructions on this part of the process have been added!